xarray/plot/utils.py

```diff
@@ -1475,17 +1474,23 @@ def _calc_widths(self, y: T | None) -> T | None:
 
         return widths
 
-    def _indexes_centered(self, x: T) -> T | None:
+    @overload
+    def _indexes_centered(self, x: np.ndarray) -> np.ndarray:
+        ...
+
+    @overload
+    def _indexes_centered(self, x: DataArray) -> DataArray:
+        ...
+
+    def _indexes_centered(self, x: np.ndarray | DataArray) -> np.ndarray | DataArray:
```

Comment:
Not sure why you didn't like the TypeVar here? But ok, the results are the same.

Reply:
I think mypy started complaining when I added typing to `.values`,  so I tried Union and finally overloads. This was one of the earliest changes though so maybe it works now when I changed ._unique_inverse ?

Reply:
Maybe one of these instances where you need to use a TypeVar bound to a Union instead?

Reply:
Maybe, I'll tests this a bit more later once I find a new bug. This works good enough for now.